ros/src/computing/perception/detection/fusion_tools/packages/range_vision_fusion/src/range_vision_fusion.cpp

```diff
@@ -354,6 +396,13 @@ RosRangeVisionFusionApp::FuseRangeVisionDetections(const autoware_msgs::Detected
             fused_objects.objects.push_back(range_in_cv.objects[vision_range_closest[i]]);
         }
     }
+    for(size_t i = 0; i < used_vision_detections.size(); i++)
```

Comment:
imo, I need to use the index to reference the vector. so instead of creating a new variable to hold the index I'm using a simple for instead of a for each (for simplicity)

Reply:
@amc-nu Sorry. After I wrote the following comment, I've found that this loop can't be range-based-for because in_vision_detections->objects[i] uses index information as you mentioned above.

-----

Because I wrote it, I'd like not to delete the following comment :)

I'd like to list the merits of using range-based-for, but it's just my opinion. If you like range-based-for, please use it where range-based-for can be used in the future work :)

1. In reality, index-based-for and range-based-for generate same assembly language, i.e., same performance.
2. Range-based-for hides detail of data access.  In range-based-for, we need not manage index information, i.e., `i = 0; i < used_vision_detections.size(); i++`.
3. Range-based-for hides detail of data access. We can change the data type (e.g., from vector to list) without modifying the loops if we want.
4. The number of characters we type and review can be decreased in range-based for, especially in the case where a lot of array accesses exist.
5. The reviewer can easily understand that range-based-for just iterates each elements, and
it doesn't require index information in the loop.

My choice between index-based-for and range-based-for is that

* I'll use index-based-for everywhere index information is mandatory, i.e., (1) index is used in if-statement: if(i < XXX), (2) index is used for another element: array[i + 1].
* I'll use range-based-for everywhere index information is unnecessary, i.e., each element is iterated for each iteration.

```
    for (auto used : used_vision_detection)
    {
        if (!used)
        {
            fused_objects.objects.push_back(in_vision_detections->objects[i]);
        }
    }
```